pkg/jobs/topology/topology_jobs.go

```diff
@@ -102,7 +102,7 @@ func SyncTopologyJob(ctx context.Context, t pkg.Topology) error {
 	}
 
 	existingTopology := existingJob.Context.Value("topology")
-	if existingTopology != nil && !reflect.DeepEqual(existingTopology.(v1.Topology).Spec, t.Spec) {
+	if existingTopology != nil && !reflect.DeepEqual(existingTopology.(v1.Topology).Spec, v1Topology.Spec) {
```

Comment:
comparing v1 spec with v1 spec, previously it was v1 spec and db's duty.json type